### PR TITLE
fix: wasmer memory access fault & caching

### PR DESCRIPTION
**Pull request by jost-s**
### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [x] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

**Review by ThetaSinner — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -66,6 +66,7 @@ incremental = false
 codegen-units = 16
 
 [patch.crates-io]
+holochain_wasmer_host = {git = "https://github.com/holochain/holochain-wasmer.git", branch = "refactor/reinstate-module-cache"}
```
Comment (ThetaSinner) [inline Cargo.toml:69]: The CI won't accept this unfortunately

Comment (jost-s) [reply Cargo.toml:69]: Ok, I just wanted to run the tests while there's no wasmer crate. Why doesn't that work?

**Comment by jost-s** [comment]
I've tidied up and added change logs.

**Comment by jost-s** [comment]
Waiting for wasmer crates to be released to update this PR and have it pass CI.

**Comment by abe-njama** [comment]
Pinging @thedavidmeister on this PR for awareness as we we just discussing this and the #3152

**Pull request merged**